PWGHF/HFL/Tasks/taskSingleMuon.cxx

```diff
@@ -20,127 +20,54 @@
-    VarManager::SetUseVars(histMan->GetUsedVars());
-    outputList.setObject(histMan->GetMainHistogramList());
-  }
-
-  template <uint32_t TEventFillMap, typename TEvent>
-  void runEventSel(TEvent const& event, aod::BCs const& bcs)
-  {
-    // select events
-    if (event.sel8()) {
-      VarManager::ResetValues(0, VarManager::kNEventWiseVariables);
-      VarManager::FillEvent<TEventFillMap>(event, values);
-      histMan->FillHistClass("EventBeforeCuts", values);
-
-      if (eventCut->IsSelected(values)) {
-        histMan->FillHistClass("EventAfterCuts", values);
-        eventSel(1);
-      } else {
-        eventSel(0);
-      }
-    } else {
-      eventSel(0);
-    }
-  }
-
-  void processEvent(MyCollisions::iterator const& event, aod::BCs const& bcs)
-  {
-    runEventSel<gEventFillMap>(event, bcs);
-  }
-  PROCESS_SWITCH(HfTaskSingleMuonEventSelection, processEvent, "run event selection with real data", true);
+namespace ambii
+{ // for MA2T
+DECLARE_SOA_INDEX_COLUMN(MFTTrack, track);
+DECLARE_SOA_INDEX_COLUMN(AmbiguousMFTTrack, ambMFTtrack);
+} // namespace ambii
 
-  void processEventMc(MyMcCollisions::iterator const& event, aod::BCs const& bcs)
-  {
-    runEventSel<gEventFillMap>(event, bcs);
-  }
-  PROCESS_SWITCH(HfTaskSingleMuonEventSelection, processEventMc, "run event selection with MC data", false);
+DECLARE_SOA_INDEX_TABLE_USER(MA2T, MFTTracks, "MA2T", ambii::MFTTrackId, ambii::AmbiguousMFTTrackId);
+} // namespace o2::aod
 
-  void processNoEventSelection(MyCollisions::iterator const& event, aod::BCs const& bcs)
-  {
-    VarManager::ResetValues(0, VarManager::kNEventWiseVariables);
-    VarManager::FillEvent<gEventFillMap>(event, values);
-    eventSel(1);
-  }
-  PROCESS_SWITCH(HfTaskSingleMuonEventSelection, processNoEventSelection, "skip the event selection", false);
+using MyCollisions = soa::Join<aod::Collisions, aod::EvSels>;
+using MyMuons = soa::Join<aod::FwdTracks, aod::FwdTracksDCA>;
+using MyMcMuons = soa::Join<aod::FwdTracks, aod::McFwdTrackLabels, aod::FwdTracksDCA>;
+using MFTTracksExtra = soa::Join<aod::MFTTracks, aod::MA2T>; // MFT track + index of ambiguous track
+
+struct ProduceMFTAmbii {
+  // build the index table MA2T
+  Builds<aod::MA2T> idx;
+  void init(InitContext const&) {}
+  // Additionnal methods provided:
+  // mfttrack.has_ambMFTtrack()
+  // mfttrack.ambMFTtrack()
 };
 
 struct HfTaskSingleMuonSelection {
-  Configurable<std::string> muonCuts{"muonCuts", "muonQualityCuts", "muon selection"};
-  Configurable<bool> fillLiteHist{"fillLiteHist", true, "fill lite version muon histograms"};
+  Configurable<uint8_t> trkType{"trkType", 0, "Muon track type, validate values are 0, 1, 2, 3 and 4"};
+  Configurable<uint8_t> mcMaskSelection{"mcMaskSelection", 0, "McMask for correct match, validate values are 0 and 128"};
+  Configurable<float> etaLow{"etaLow", -3.6, "Eta acceptance low edge"};
+  Configurable<float> etaUp{"etaUp", -2.5, "Eta acceptance up edge"};
```

Comment:
The opposite of low is high, not up. But you should use `Min` and `Max` instead.